web-api-shared/src/elastic.rs

```diff
@@ -104,9 +115,25 @@ impl Client {
                 .with_segments([&index_name])
                 .into(),
             client: reqwest::ClientBuilder::new().timeout(timeout).build()?,
+            retry_policy,
         })
     }
 
+    pub async fn retry<T, E, F>(
+        &self,
+        error_filter: impl Fn(&E) -> bool,
+        code: impl FnMut() -> F,
+    ) -> Result<T, E>
+    where
+        F: Future<Output = Result<T, E>>,
+        E: Display,
+    {
+        let policy = ExponentialJitterRetryPolicy::new(self.retry_policy.clone())
+            .with_retry_filter(error_filter);
```

Comment:
The retry filter is call context specific and might be affected by the generic parameters of the functions it's called in, so in turn we always want to configure it "around the places" where we use the policy.